parser_library/src/lsp/lsp_context.cpp

```diff
@@ -906,11 +926,11 @@ std::string lsp_context::find_hover(const symbol_occurrence& occ, macro_info_ptr
                 auto it = m_macros.find(occ.opcode);
                 assert(it != m_macros.end());
 
-                return hover_for_macro(*it->second);
+                return prefix_using(hover_for_macro(*it->second));
```

Comment:
There's something going wrong with the following code when hover is being done over `SQLSECT` - It seems to be linked to the `hover_for_macro` call:

```

         AIF (L'X EQ 4).SKIP1
         SAM31
.SKIP1   ANOP

         AGO .SKIP2
X        DS F
         MNOTE 2,                                                      X
               'aAAA'
         MNOTE 2,                                                      X
               'aAAA'
.SKIP2   ANOP

         SQLSECT SAVE

         END
 ```

![image](https://github.com/BroadcomMFD-DevOps/che-che4z-lsp-for-hlasm/assets/99467904/473f4a44-edb4-4ee0-8120-07c265b3ab4a)


btw line 5 contains the empty line fade diagnostic

Reply:
ok, so this seems unrelated. It looks like we are not holding onto the texts for virtual files for some reason?

Reply:
Not sure about the reason but I've made a note and can (probably) take a look at this later. Closing for now

Reply:
I see the reason 🤦